connector-j-6/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java

```diff
@@ -63,6 +67,19 @@ public Socket connect(String host, int portNumber, Properties props, int loginTi
     return this.socket;
   }
 
+  private List<String> listIpTypes(String cloudSqlIpTypes) {
```

Comment:
Have you considered moving IpTypes into the SSLSocketFactory instead? Since it isn't needed until then, it would reduce the need to have this function in every class.